core/persistence-jpa/src/main/java/org/apache/syncope/core/persistence/jpa/dao/JPAAccessTokenDAO.java

```diff
@@ -115,6 +115,16 @@ public AccessToken save(final AccessToken accessToken) {
         return entityManager().merge(accessToken);
     }
 
+    @Override
+    @Transactional(rollbackFor = Throwable.class)
+    public void merge(final AccessToken accessToken) {
+        AccessToken existing = findByOwner(accessToken.getOwner());
+        if (existing != null) {
+            entityManager().remove(existing);
+        }
+        entityManager().persist(accessToken);
```

Comment:
replace with `save(accessToken)` and make this method return `AccessToken` as `save()` does.

Reply:
Also, replace `entityManager().remove(existing)` with `delete(existing)`, thanks.